### PR TITLE
Decorator a class

### PR DESCRIPTION
**Pull request by chukwuemeka1234**
**Hi there** 👋
### The branch decorate-a-class has the following features to be merged into the development branch:

- [ ] Created a class Corrector with a method correct_name that does the following:
- Make sure the first letter of the given name is a capital letter.
- Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.
- [ ] Set an instance of Corrector in Person on initialization.

- [ ] In Person add a method validate_name that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Corrector
```
Comment (rloterh) [inline corrector.rb]: - [ ] Nameable interface is missing. Kindly ensure to create a `Nameable` class with a method 'correct_name' rasing an error, like the following:
```ruby
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```

File: person.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require_relative 'corrector'
+
 class Person
```
Comment (rloterh) [inline person.rb]: - [ ] `Person` class should inherits from `Nameable` class. It should also have a method `correct_name`:
```ruby
class Person < Nameable
  .
  .

    def correct_name
       name
    end
  .
  .
end
```


Kindly ensure to improve upon this aspect.

File: corrector.rb (hunk 1 above)
Comment (rloterh) [inline corrector.rb]: Kindly ensure to provide a _Base Decorator_ which:

- [ ] Check if the `Decorator` class inherits from Namable.
- [ ] Check if in the constructor a namable object from params is assigned to an instance variable.
- [ ] Check of the correct_name method that returns the result of the correct_name method of the @namable is implemented.

The decorator base should look like the following:
```ruby
class Decorator < Nameable
    attr_accessor :nameable
  
    def initialize(nameable)
      @nameable = nameable
    end
  

    def correct_name
      @nameable.correct_name
    end
end
```

Comment (rloterh) [inline corrector.rb]: Kindly ensure to add decorator classes.
- [ ] `CapitalizeDecorator` class, should look similar to:

```ruby
# Inherits from Decorator
class CapitalizeDecorator < Decorator
    # Capitalize the first letter of the given name.
    def correct_name
      @nameable.correct_name.capitalize
    end
end
```

- [ ] `TrimmerDecorator` class, should look similar to:

```ruby
# Inherits from Decorator
class TrimmerDecorator < Decorator
    # The name has a maximum of 10 characters. If it's longer it should trim the word.
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Chukwuemeka, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @chukwuemeka1234,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**